proxy.go

```diff
@@ -0,0 +1,97 @@
+package gimlet
+
+import (
+	"math/rand"
+	"net/http"
+	"net/http/httputil"
+	"strings"
+
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/mongodb/grip/message"
+)
+
+// ProxyOptions describes a simple reverse proxy service that can be
+// the handler for a route in an application. The proxy implementation
+// can modify the headers of the request. Requests are delgated to
+// backends in the target pool
+type ProxyOptions struct {
+	HeadersToDelete   []string
+	HeadersToAdd      map[string]string
+	RemotePrefix      string
+	StripSourcePrefix bool
+	TargetPool        []string
+}
+
+// Validate checks the default configuration of a proxy configuration.
+func (opts *ProxyOptions) Validate() error {
+	if !strings.HasPrefix(opts.RemotePrefix, "/") {
+		opts.RemotePrefix = "/" + opts.RemotePrefix
+	}
+
+	catcher := grip.NewBasicCatcher()
```

Comment:
Seems slightly silly to have a catcher for a single error.

Reply:
yeah, but it makes it easy to add more things.